### ISSUE TITLE
Some dots (gouge, insecticide and chir's link) counting as 2 debuff for death mark

### ISSUE THREAD

**Issue by Jikaable** [TeamMoonstorm/Starstorm2#423]
Hey,
As stated in the title those three dots proc death mark too soon, specifically proccing it on the second instance of damage that has a proc coefficient.
After doing a bit of research I stumbled upon that deprecated mod (the game doesn't even launch when it's installed) and its description that seems to have found the root cause of the issue :
https://thunderstore.io/package/Nebby/ConfigurableDeathMark/

Basically it is possible that some dots are setup as dots AND as debuffs and death marks checks both separately.

**Comment by Hevia** [comment]
I believe this has to do with how we write our debuffs/dots in code (both dot and debuff are “buffdefs” iirc)

So this is expected vanilla behavior, but we can look into polishing the experience a bit so it functions how most people assume it would and I gotta double check my statement there. I’m pretty sure Gouge is setup that way but not sure about the other two

**Comment by Jikaable** [comment]
> I believe this has to do with how we write our debuffs/dots in code (both dot and debuff are “buffdefs” iirc)
> 
> So this is expected vanilla behavior, but we can look into polishing the experience a bit so it functions how most people assume it would and I gotta double check my statement there. I’m pretty sure Gouge is setup that way but not sure about the other two

While I understand that it is "expected vanilla behavior" based on how the dots are setup in starstorm 2 (not trematode surprisingly) I wouldn't say that it is how it's "supposed" to work. None of the dots in the base game count for two debuffs regarding death mark.
Thanks for acknowledging my reports anyway, and thank you for your work on the mod and its issues

**Comment by Jikaable** [comment]
> This is because chances are SS2's dot BuffDefs are mamrked as Debuffs, deathmark count increases with DOT count AND debuff count.
> 
> This can be easily fixed on our end, or if you want, try out my own fix with [Configurable DeathMark](https://thunderstore.io/package/Nebby/ConfigurableDeathMark/) (afaik it still works post SOTV)

Hi, yeah I know about your mod, I even linked to it in the main post. After further testing the game does launch with only it (and its dependencies) enabled but sadly with R2API_Core enabled the game gets stuck during loading at 100%.
Here's the log file : 
[LogOutput.log](https://github.com/TeamMoonstorm/Starstorm2/files/14088206/LogOutput.log)

**Comment by Hevia** [comment]
@Nebby1999 Just double checking here. Did you fix this with the new BuffDef class in MSU 2.0?

**Comment by Nebby1999** [comment]
> @Nebby1999 Just double checking here. Did you fix this with the new BuffDef class in MSU 2.0 (not yet released for those reading)?

yes, DotBuffDefs as theyre called force the isDebuff boolean to be false.

**Comment by Hevia** [comment]
Ok cool

@Jikaable This will be fixed whenever 0.7 releases since thats our timeline for SS2 upgrading to MSU 2.0
